### PR TITLE
EMB-1718: centralize env validation behind a single Zod schema

### PR DESCRIPTION
**Pull request by adam-edison**
## Summary

Replaces the scattered env-var checks (per-service validators, ad-hoc `process.env.X!` reads in components, and a presence-only build script) with a single Zod schema at `src/shared/config/EnvironmentSchema.ts` and a `Configuration` singleton that parses the env once at boot. A misconfigured deploy now fails with one aggregated error listing every missing or malformed var instead of surfacing one var at a time when the contact form is first submitted.

[EMB-1718](https://linear.app/ember-and-tide/issue/EMB-1718/config-centralize-env-var-validation-at-startup-with-single-zod-schema)

## Changes

- Adds `src/shared/config/EnvironmentSchema.ts` with all env vars split into a server-only schema and a `NEXT_PUBLIC_` client-safe subset (URLs, emails, integers with bounds, boolean coercion, ms-compatible window regex).
- Adds `Configuration` singleton (lazy parse, aggregated error reporting, `forTesting` factory, `reset` for test isolation).
- Adds `ClientConfiguration` accessor for the `NEXT_PUBLIC_` subset so React components import a validated, typed config.
- Migrates every consumer to read through the config layer: `Logger`, `social-links`, `RateLimiterDataStore`, `RateLimiter`/`ContactRateLimiter`, `TurnstileClient`, `EmailService`, `ContactForm`, and `CommandPalette`.
- Drops the old `EmailServiceConfigurationValidator`, `EmailServiceConfigurationFactory`, and their unit tests; the schema and Configuration are the single source of truth.
- `TurnstileClient.fromEnv()` and `EmailService.fromEnv()` now return their clients directly (the schema makes the construction-failure paths unreachable). `ContactFormProcessor.fromEnv()` becomes synchronous; `/api/contact` and `/api/email-service-check` are simplified to match.
- Rewrites `scripts/check-env.ts` as a thin tsx wrapper that runs the same Zod schema, so the build hook and runtime share validation.
- Adds a side-effect setup file (`tests/setup/configurationForTesting.ts`) that primes Configuration and ClientConfiguration with valid test defaults before any module that reads env vars is imported.
- Updates `CLAUDE.md` and `README.md` env-var sections so they match what the schema actually requires today (adds `REDIS_PREFIX`, documents `SEND_EMAIL_ENABLED`, marks rate-limit vars as required).

## Special Notes

- `NODE_ENV` is part of the schema even though the ticket scope listed it under "scattered raw reads". Logger needs it at module-construction time, so adding it to the schema keeps the "no `process.env` reads outside Configuration" rule intact for app code.
- Sentry, email-quota, and `LOG_LEVEL` / `LOG_TIMEZONE` vars listed in the ticket scope are intentionally **not** in the schema yet — they belong to gap tickets that have not landed; the schema will grow when they do. The schema is the single source of truth for vars the app reads today.
- The `_app` chunk grew by ~13 kB because Zod is now imported on the client side via `ClientConfiguration` (previously Zod only landed in the lazily-loaded contact-form chunk via `ContactFormValidator`). Worth a follow-up if the size matters; outside the scope of this ticket.

### 1. Unit / component / integration tiers pass locally

- [x] Run `npm run test:unit`
- [x] All 128 tests pass
- [x] `EnvironmentSchema.unit.test.ts` covers required-vs-optional, URL/email validators, integer bounds, boolean coercion, and rate-limit window regex
- [x] `clientConfig.unit.test.ts` covers the same surface for the client subset
- [x] Run `npm run test:component`
- [x] All 15 component tests pass (Turnstile + ContactForm)
- [x] Run `npm run test:integration`
- [x] All 8 integration tests pass against real Upstash / Cloudflare endpoints

### 2. Build hook fails loudly when an env var is missing

- [x] Temporarily unset `RESEND_API_KEY` in your local environment file
- [x] Run `npm run build`
- [x] Build aborts with `Build failed: environment validation errors`
- [x] Output lists `RESEND_API_KEY` on its own line
- [x] Restore the env var
- [x] Run `npm run build`
- [x] Build prints the success line `All environment variables validated against EnvironmentSchema` before `next build` runs
- [x] `next build` completes successfully

### 3. Runtime behavior unchanged from a user's perspective

- [x] Run `npm run dev`
- [x] Open `http://localhost:3000`
- [x] Header social icons render with the correct GitHub and LinkedIn URLs (no broken `undefined` hrefs)
- [x] Open the command palette (Cmd/Ctrl+K)
- [x] Selecting **GitHub** opens the configured GitHub URL in a new tab
- [x] Selecting **LinkedIn** opens the configured LinkedIn URL in a new tab
- [x] Selecting **View Source Code** opens the configured repo URL in a new tab
- [x] Visit `/contact`
- [x] Contact form renders the Turnstile widget (no "Contact form is not available" error)
- [x] `GET /api/email-service-check` returns `{"status":"healthy"}`
- [x] Submit the contact form with a valid Turnstile token
- [x] Form succeeds and rate-limit headers (`X-RateLimit-*`) appear in the response

### 4. Schema is the single source of truth

- [x] Run `rg "process.env" src/`
- [x] Output only matches inside `src/shared/config/Configuration.ts`, `src/shared/config/clientConfig.ts`, and `src/shared/Logger.ts` (documented exception: `NODE_ENV` is statically inlined by Next.js at build time for both bundles, so the direct read cannot fail at runtime)
- [x] No app code outside the config layer reads `process.env` directly, except the documented `NODE_ENV` exception in `src/shared/Logger.ts`

### 5. Automated coverage

- [x] `tests/unit/shared/config/EnvironmentSchema.unit.test.ts`
  - [ ] parses a fully valid environment with type coercion
  - [ ] aggregates every missing required field into one error
  - [ ] treats `SEND_EMAIL_ENABLED` as optional defaulting to false
  - [ ] coerces `SEND_EMAIL_ENABLED` only when value equals literal `"true"`
  - [ ] rejects malformed URLs / emails / non-numeric ints / out-of-bounds ints
  - [ ] accepts ms-compatible window strings (`500 ms`, `30 s`, `10 m`, `1 h`, `1 d`, `5m`)
  - [ ] rejects malformed window strings
  - [ ] defaults `NODE_ENV` to development; rejects unknown values
  - [ ] `ClientEnvironmentSchema` parses just the `NEXT_PUBLIC_*` subset
  - [ ] `ServerOnlyEnvironmentSchema` does not require `NEXT_PUBLIC_*` fields
- [x] `tests/unit/shared/config/clientConfig.unit.test.ts`
  - [ ] mirrors the Configuration test surface for the `NEXT_PUBLIC_*` subset
- [x] `tests/unit/features/contact/EmailService.unit.test.ts`
  - [ ] builds an EmailService whose configuration mirrors Configuration values
  - [ ] exposes `sendEmailEnabled` correctly for `undefined`, `"false"`, and `"true"`
- [x] `tests/unit/features/contact/TurnstileClient.unit.test.ts`
  - [ ] builds a client using the secret from Configuration
  - [ ] retains all previous fail-closed behaviors for empty token, network failure, non-OK status, rejected response, malformed JSON, and timeouts
- [x] `tests/integration/features/contact/TurnstileClient.integration.test.ts`
  - [ ] always-pass secret returns success against real Cloudflare
  - [ ] always-fail secret fails closed with `verification rejected`
- [x] `tests/integration/features/rate-limiting/RateLimiter.integration.test.ts`
  - [ ] all 6 sliding-window scenarios still pass with the new Configuration-driven Redis client

### CODE REVIEW

**Review by adam-edison — COMMENTED** [review]
## AI Code Review

Centralizes env validation into a clean Zod schema and Configuration singleton — solid execution. **One high-priority concern**: `Logger.create()` runs at module init and is imported by 5 client components, so the full server schema (with `RESEND_API_KEY`, `TURNSTILE_SECRET_KEY`, etc.) is shipped to and executed in the browser bundle, where validation will throw on hydration.

## Action Items

### High

- [⏭️] **[Architecture] Decouple client-imported Logger from Configuration** `src/shared/Logger.ts:11`
  - **Problem:** `Logger.create()` is called at module init (line 84: `export const logger = Logger.create()`) and invokes `Configuration.get()`, but Logger is transitively imported by 5 client components (ContactForm, ContactFormInner, ContactFormErrorBoundary, ContactFormLazy, SplitTextLazy), so the full server `EnvironmentSchema` is parsed in the browser where `process.env` lacks the required server-only vars.
  - **Consequence:** Client hydration of `/contact` (and any page importing those components) would throw at runtime; the build masks it because SSR runs in Node and tests pre-cache via `tests/setup/configurationForTesting.ts`. Verified server-only var names appear in `.next/static/chunks/pages/{_app,contact,index}-*.js`.
  - **Fix:** Reintroduce a NODE_ENV-only accessor in Logger (read `process.env.NODE_ENV` directly — Next.js inlines it on the client) or expose a client-safe `RuntimeEnv.nodeEnv()` that does not import `EnvironmentSchema`; verify by loading `/contact` in a real browser.
  - **Ticket:** https://linear.app/ember-and-tide/issue/EMB-1926/decouple-logger-from-configuration-to-stop-server-schema-reaching-the

### Medium

- [⏭️] **[Security] Stop leaking server-only env-var names into the client bundle** `src/shared/Logger.ts:11`
  - **Problem:** Because Logger imports `Configuration`, the bundler ships every `EnvironmentSchema` field name (`RESEND_API_KEY`, `TURNSTILE_SECRET_KEY`, `UPSTASH_REDIS_REST_URL/TOKEN`, etc.) into the client bundle as schema literals.
  - **Consequence:** Discloses the application's server-only env-var inventory to anyone inspecting the JS, broadening reconnaissance for misconfiguration probing and giving exact secret-key naming for spear-phishing operators.
  - **Fix:** Same as the High finding — remove Configuration from Logger's import graph so no server schema reaches the client.
  - **Ticket:** https://linear.app/ember-and-tide/issue/EMB-1926/decouple-logger-from-configuration-to-stop-server-schema-reaching-the

- [x] **[Architecture] Eliminate duplicated NEXT_PUBLIC_* test defaults across Configuration and clientConfig** `src/shared/config/Configuration.ts:6` ([1052f07](https://github.com/adam-edison/adam-edison-dot-com/commit/1052f07ae48334fc752a53ce7a4808c4050b3704))
  - **Problem:** `TEST_DEFAULTS` in `Configuration.ts` and `TEST_DEFAULTS` in `clientConfig.ts` repeat the same four `NEXT_PUBLIC_*` literal values; if a future engineer tightens one (e.g. the LinkedIn URL pattern) the two will drift silently.
  - **Consequence:** Drift between server-side and client-side test defaults could let a test pass on one surface and fail on the other for unrelated reasons, or hide bugs entirely; also violates the PR's stated single-source-of-truth goal.
  - **Fix:** Have `Configuration.ts` spread `clientConfig.TEST_DEFAULTS` (or vice versa) so the `NEXT_PUBLIC_*` values are defined once.

- [⏭️] **[Architecture] Harden the cached singleton against test/prod cross-contamination** `src/shared/config/Configuration.ts:31`
  - **Problem:** `Configuration` uses static mutable state (`private static cached`) that mutates via `forTesting()`/`reset()`; a test that forgets to call `reset()` after `forTesting` leaves the cache populated with test defaults.
  - **Consequence:** Hidden inter-test ordering coupling; if production code ever runs in the same V8 isolate as a test (e.g. via a misordered import in e2e setup) it would silently use test secrets without error.
  - **Fix:** Move `forTesting`/`reset` into a separate test-only module that hooks into Configuration's internal cache via a narrow API, and document the cache as request-cycle scoped.
  - **Ticket:** https://linear.app/ember-and-tide/issue/EMB-1927/move-configuration-fortestingreset-and-test-defaults-into-a-test-only

- [x] **[Tests] Remove or adapt the dead `vi.stubEnv` calls in the ContactForm component test** `tests/component/features/contact/components/ContactForm.component.test.tsx:61` ([abb9fbf](https://github.com/adam-edison/adam-edison-dot-com/commit/abb9fbf001004a24466665fd49b49a5072f55a42))
  - **Problem:** The test stubs `NEXT_PUBLIC_TURNSTILE_SITE_KEY`/`TURNSTILE_SECRET_KEY`/`SEND_EMAIL_ENABLED` via `vi.stubEnv`, but `ClientConfiguration.get()` is already cached by `tests/setup/configurationForTesting.ts` before the test runs.
  - **Consequence:** The stubs are silently ineffective; existing tests pass only because they don't assert the specific siteKey value, and a future test depending on changing the cached value would fail in confusing ways.
  - **Fix:** Replace `stubEnv` with `ClientConfiguration.forTesting({ NEXT_PUBLIC_TURNSTILE_SITE_KEY: 'test-site-key' })` and reset in `afterEach`.

- [⏭️] **[Tests] Restore meaningful coverage of `TurnstileClient.fromEnv`** `tests/unit/features/contact/TurnstileClient.unit.test.ts:18`
  - **Problem:** The `fromEnv` test only asserts `expect(client).toBeInstanceOf(TurnstileClient)`, which is a tautology given the typed return signature; the deleted tests verified the secret was actually read from the environment and propagated to `verifyToken`.
  - **Consequence:** Coverage regressed; a future bug where `fromEnv` reads the wrong key (e.g. `RESEND_API_KEY` instead of `TURNSTILE_SECRET_KEY`) would not be caught.
  - **Fix:** Extend an existing `verifyToken` mock-fetch test to inspect the request body for the configured secret value, asserting it matches what `forTesting` set.
  - **Skipped:** Already covered by the `passes secret, response, and remoteIp to Turnstile API` test in the same file, which asserts `body.toContain('secret=test-secret')` after `Configuration.forTesting({ TURNSTILE_SECRET_KEY: 'test-secret' })` — a swap to the wrong key would fail that assertion.

### Low

- [⏭️] **[Security] Mark `TEST_DEFAULTS` as obviously test-only** `src/shared/config/Configuration.ts:6`
  - **Problem:** `TEST_DEFAULTS` contains realistic-looking placeholder secrets (`test-resend-key`, `test-turnstile-secret`, `test-token`) shipped in the production build alongside `Configuration`.
  - **Consequence:** A future regression that calls `forTesting()` outside a test would let the app boot with bogus credentials and hit Resend/Cloudflare/Upstash with strings that don't trigger validation errors but silently break the relevant feature in production.
  - **Fix:** Move `TEST_DEFAULTS` into a `tests/`-only file, or guard `forTesting` with `assert(process.env.NODE_ENV === 'test')` that throws if invoked outside tests.
  - **Ticket:** https://linear.app/ember-and-tide/issue/EMB-1927/move-configuration-fortestingreset-and-test-defaults-into-a-test-only

- [x] **[Performance] Defer `ClientConfiguration.get` until first read in social-links** `src/features/navigation/social-links.ts:3` ([50b2bab](https://github.com/adam-edison/adam-edison-dot-com/commit/50b2bab1ab5423cb983a1fd3f4ca76932d4874be))
  - **Problem:** `const clientConfig = ClientConfiguration.get();` runs synchronously on module import, forcing schema validation into the import graph for every client page that links to social-links.ts.
  - **Consequence:** Adds eager-validation cost during initial page load and prevents tests using `vi.stubEnv` on `NEXT_PUBLIC_*` values from taking effect (cache populates before stub runs).
  - **Fix:** Replace the module-level call with a getter or read inside the consuming component, mirroring how `CommandPalette` already reads on render.

- [⏭️] **[Performance] Stabilise siteKey computation in ContactForm** `src/features/contact/components/ContactForm.tsx:13`
  - **Problem:** `ClientConfiguration.get()` is called on every render of `ContactForm`, even though the cached value never changes.
  - **Consequence:** Negligible per-render cost; primarily an intent/readability issue.
  - **Fix:** Compute `siteKey` once at module scope or with `useMemo`, mirroring the codebase's pattern for stable env-derived constants.
  - **Skipped:** YAGNI. The reviewer flagged the cost as negligible and the cache makes repeated reads free; module-scope evaluation would couple import order to test setup, which is what other items in this review are explicitly trying to avoid.

- [x] **[Architecture] Tighten `RATE_LIMIT_WINDOW_PATTERN` to reject zero counts** `src/shared/config/EnvironmentSchema.ts:3` ([18a4202](https://github.com/adam-edison/adam-edison-dot-com/commit/18a4202a3d4288b33ce41336c945b2bbd963a999))
  - **Problem:** The `^\d+\s*(ms|s|m|h|d)$` regex accepts `0 s`, `0 m`, etc., which `@upstash/ratelimit` treats as a zero-length sliding window — an effectively-disabled limiter.
  - **Consequence:** A misconfigured deploy with `RATE_LIMIT_WINDOW=0 s` would silently disable per-IP rate limiting at runtime instead of failing the build, defeating the protection.
  - **Fix:** Change the leading group to `[1-9]\d*` and add a unit test pinning the rejection.

- [⏭️] **[Architecture] Tighten `SEND_EMAIL_ENABLED` boolean coercion** `src/shared/config/EnvironmentSchema.ts:22`
  - **Problem:** `optionalBoolFromString` silently coerces every value other than the literal `'true'` (including `'TRUE'`, `'1'`, `'yes'`) to false.
  - **Consequence:** An operator who sets `SEND_EMAIL_ENABLED=TRUE` gets a silent off-state with no error — exactly the misconfiguration the schema is meant to prevent.
  - **Fix:** Either accept a broader true-set (`['true','1','yes']`) and reject everything else as an error, or keep the strict policy but throw on unrecognised non-empty values rather than coerce.
  - **Ticket:** https://linear.app/ember-and-tide/issue/EMB-1928/tighten-send-email-enabled-boolean-coercion-to-error-on-unrecognized

- [x] **[Style] Rename the `parseEnvironment` parameter from `env`** `src/shared/config/Configuration.ts:33` ([066db1e](https://github.com/adam-edison/adam-edison-dot-com/commit/066db1e89a8b3cd231fa8a6b20c6f3acb60725e5))
  - **Problem:** The local helper `parseEnvironment(env: Record<...>)` shadows the `env` constant used inside callers (e.g. `EmailService.fromEnv` calls its local var `env` for the parsed Configuration object).
  - **Consequence:** Reduces grep-ability and forces readers to reason about which `env` is which (raw `process.env` vs parsed Environment).
  - **Fix:** Rename the parameter to `rawEnv` or `processEnv` to signal the unparsed input.

- [x] **[Style] Use `config` (not `env`) for the parsed Configuration in EmailService.fromEnv** `src/features/contact/EmailService.ts:43` ([066db1e](https://github.com/adam-edison/adam-edison-dot-com/commit/066db1e89a8b3cd231fa8a6b20c6f3acb60725e5))
  - **Problem:** Inside `EmailService.fromEnv`, the local name `env` is used for the parsed Configuration object, while `ContactRateLimiter` and `RateLimiterDataStore` use `config` for the same shape.
  - **Consequence:** Intra-PR style drift; readers may believe `env` is `process.env` when it is the validated `Environment`.
  - **Fix:** Rename to `config` to match the rest of the migrated call sites.

- [⏭️] **[Style] Document the removal of the missing-siteKey early return in ContactForm useEffect** `src/features/contact/components/ContactForm.tsx:15`
  - **Problem:** The PR removed the `siteKey` guard inside `useEffect` but kept `useEffect(() => { checkServerConfig(); }, []);` with an empty array; the component now unconditionally fetches `/api/email-service-check` on mount.
  - **Consequence:** Future readers won't realise the component changed from "fetch only when siteKey is set" to "fetch always," and the comment trail does not explain why.
  - **Fix:** Either restore the early-return invariant (now redundant since the schema enforces siteKey presence) or drop a one-line comment noting the schema guarantees siteKey at boot.
  - **Skipped:** Repo style avoids comments that restate behavior; the schema's `NEXT_PUBLIC_TURNSTILE_SITE_KEY` requirement makes the invariant obvious from `EnvironmentSchema.ts`. Restoring the guard would re-introduce dead defensive logic now that the schema enforces siteKey at boot.

- [⏭️] **[Style] Replace the manual `readClientEnv` whitelist with a typed iteration** `src/shared/config/clientConfig.ts:32`
  - **Problem:** `readClientEnv` hardcodes the four `NEXT_PUBLIC_*` keys; adding a fifth means editing both the schema and this function with nothing forcing them in sync. (Note: explicit literals are required for Next.js inlining — see the comment below.)
  - **Consequence:** A future engineer who adds `NEXT_PUBLIC_FOO` to `ClientEnvironmentSchema` but forgets `readClientEnv` will silently get `undefined` and a confusing schema error.
  - **Fix:** Iterate over `Object.keys(ClientEnvironmentSchema.shape)` and read each with bracket notation; if Next.js inlining requires literal access, add a comment that explicitly calls this constraint out.
  - **Skipped:** Next.js inlines `process.env.FOO` only when the access is a literal property read, so iterating over schema keys would defeat the inlining and break NEXT_PUBLIC delivery on the client. The reviewer's note acknowledges the constraint, and the repo's no-comments rule rules out the suggested explanatory comment.

- [x] **[Style] Extract the duplicated `buildEnvironmentError` helpers** `src/shared/config/Configuration.ts:53` ([bc04c69](https://github.com/adam-edison/adam-edison-dot-com/commit/bc04c69fde5a0d58694a8d4f91a57f750e117583))
  - **Problem:** `buildEnvironmentError` (Configuration) and `buildClientEnvironmentError` (clientConfig) differ only in the leading message string; the issue-formatting logic is duplicated.
  - **Consequence:** Future improvements (adding the offending value, colorising) must be made in two places.
  - **Fix:** Extract a single `formatZodIssues(prefix, issues)` helper into `EnvironmentSchema.ts` and call from both classes.

- [x] **[Tests] Replace the self-mirror assertion in the rate-limit-window test** `tests/unit/shared/config/EnvironmentSchema.unit.test.ts:1467` ([68b4f97](https://github.com/adam-edison/adam-edison-dot-com/commit/68b4f973f431cce90eb1b109a7372e38a7e711e3))
  - **Problem:** The "accepts ms-compatible window strings" test maps over `variants` then asserts against `variants.map((window) => ({ window, success: true }))` — the same source array transformed identically.
  - **Consequence:** The construction is fragile and hides the actual expected values; readers cannot eyeball what the test is pinning.
  - **Fix:** Hard-code the expected array (e.g. `[{ window: '500 ms', success: true }, ...]`) so the assertion is explicit.

- [x] **[Tests] Cover the validation-failure path in `EmailService.fromEnv`** `tests/unit/features/contact/EmailService.unit.test.ts` ([1afcb30](https://github.com/adam-edison/adam-edison-dot-com/commit/1afcb30cb38220d34b19453c28dedce27f6265d6))
  - **Problem:** The PR removed the `EmailServiceConfigurationValidator` tests that asserted invalid env values surface a useful error; the new test file only covers happy-path `SEND_EMAIL_ENABLED` variations.
  - **Consequence:** No regression test guarantees an invalid configuration short-circuits before `new EmailService(config)` is reached; a future change that swallows the Configuration error could silently let a partial EmailService load.
  - **Fix:** Add one test that uses `Configuration.forTesting({ FROM_EMAIL: 'bad' })` (via the test factory) and asserts `EmailService.fromEnv()` throws with a message naming `FROM_EMAIL`.

- [x] **[Tests] Add a test for the production `Configuration.get()` path** `tests/unit/shared/config/Configuration.unit.test.ts` ([dbf373a](https://github.com/adam-edison/adam-edison-dot-com/commit/dbf373ad033c5d58f359b0ecef939d865f185f79))
  - **Problem:** The unit tests cover `forTesting`/`get`/`reset` interactions only via `forTesting`; they never exercise `Configuration.get()` parsing real `process.env` values.
  - **Consequence:** The production read path is untested by unit tests; only the boundary build script and e2e indirectly exercise it.
  - **Fix:** Add a test that stubs `process.env`, calls `Configuration.reset()` then `Configuration.get()`, and asserts the parsed result mirrors the stubs.

- [x] **[Tests] Restore explicit `Configuration.reset()` cleanup in TurnstileClient integration test** `tests/integration/features/contact/TurnstileClient.integration.test.ts:21` ([2ea4445](https://github.com/adam-edison/adam-edison-dot-com/commit/2ea4445c3fda7875be49ec0a336282a7190c96c9))
  - **Problem:** `buildClientWith` mutates `process.env.TURNSTILE_SECRET_KEY` and calls `Configuration.reset()`, but there is no `afterEach` that restores the original env or resets after the test.
  - **Consequence:** Hidden inter-test ordering dependency; any later integration test added to this file could silently inherit the always-pass or always-fail Turnstile secret.
  - **Fix:** Add an `afterEach` that restores the original `TURNSTILE_SECRET_KEY` and calls `Configuration.reset()`.

## Already Planned

- **Remove the now-tautological `/api/email-service-check` round-trip** `src/pages/api/email-service-check.ts:17` — covered by EMB-1775: Eliminate /api/email-service-check round-trip. The handler now calls `Configuration.get()` purely for its throw-on-misconfig side effect, but Configuration is cached at boot and `EnvironmentSchema` is enforced at build time, so it can never return anything but 200.

File: src/shared/Logger.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+
 export class Logger {
   protected formatMessage(level: string, message: string): string {
     const timestamp = new Date().toISOString();
     return `${timestamp} ${level} ${message}`;
   }
 
   static create(): Logger {
-    if (process.env.NODE_ENV === 'test') {
+    if (Configuration.get().NODE_ENV === 'test') {
```
Comment (adam-edison) [inline src/shared/Logger.ts:11]: I don't like that we just have indirection around the process.env variables.
We should have a real configuration object that we pass around and inject instead of using statics everywhere. This is just bad design.

**Review by adam-edison — COMMENTED** [review]
bad design

File: src/shared/Logger.ts (hunk 1 above)
Comment (adam-edison) [reply src/shared/Logger.ts:11]: Filed as https://linear.app/ember-and-tide/issue/EMB-1929/config-replace-configuration-static-accessor-with-injected for follow-up.

File: README.md [hunk 2]
```diff
@@ -65,12 +65,14 @@ FROM_EMAIL=your-contact@email.com
 TO_EMAIL=your-inbox@email.com
 EMAIL_SENDER_NAME=Personal Website Contact Form
 EMAIL_RECIPIENT_NAME=Your Name
+SEND_EMAIL_ENABLED=true
```
Comment (adam-edison) [inline README.md]: We shouldn't have this level of detail in the readme it should just be referred to the example environment file. Get rid of all these specific settings.

We just need notes on the environment variables themselves in the example file.

File: tests/unit/shared/config/Configuration.unit.test.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import { describe, it, expect, beforeEach, afterEach, vi } from 'vitest';
```
Comment (adam-edison) [inline tests/unit/shared/config/Configuration.unit.test.ts]: I hate that we are testing configuration. It seems completely unnecessary. Is there a reason we need these tests?

**Review by adam-edison — COMMENTED** [review]
I hate this ticket

File: README.md (hunk 2 above)
Comment (adam-edison) [reply README.md]: Addressed in [75175e5](https://github.com/adam-edison/adam-edison-dot-com/commit/75175e5d72e94b077616ee26571f292ffa7bc045)

File: tests/unit/shared/config/Configuration.unit.test.ts (hunk 3 above)
Comment (adam-edison) [reply tests/unit/shared/config/Configuration.unit.test.ts]: Addressed in [581bd37](https://github.com/adam-edison/adam-edison-dot-com/commit/581bd374e7a7f1c30ab6de0ba5e564fde75b97dd)
